awx/main/models/unified_jobs.py

```diff
@@ -1270,6 +1298,9 @@ def cancel(self, job_explanation=None, is_chain=False):
                 if self.status in ('pending', 'waiting', 'new'):
                     self.status = 'canceled'
                     cancel_fields.append('status')
+                if self.status == 'running' and not self.actually_running:
```

Comment:
I would prefer `elif`, otherwise, `cancel_fields` could have `['status', 'status']`, and we'd be assuming Django will not care.